rust/deny-check/deny.toml

```diff
@@ -0,0 +1,212 @@
+# This template contains all of the possible sections and their default values
+
+# Note that all fields that take a lint level have these possible values:
+# * deny - An error will be produced and the check will fail
+# * warn - A warning will be produced, but the check will not fail
+# * allow - No warning or error will be produced, though in some cases a note
+# will be
+
+# The values provided in this template are the default values that will be used
+# when any section or field is not specified in your own configuration
+
+# If 1 or more target triples (and optionally, target_features) are specified,
+# only the specified targets will be checked when running `cargo deny check`.
+# This means, if a particular package is only ever used as a target specific
+# dependency, such as, for example, the `nix` crate only being used via the
+# `target_family = "unix"` configuration, that only having windows targets in
+# this list would mean the nix crate, as well as any of its exclusive
+# dependencies not shared by any other crates, would be ignored, as the target
+# list here is effectively saying which targets you are building for.
+targets = [
+    # The triple can be any string, but only the target triples built in to
+    # rustc (as of 1.40) can be checked against actual config expressions
+    #{ triple = "x86_64-unknown-linux-musl" },
+    # You can also specify which target_features you promise are enabled for a
+    # particular target. target_features are currently not validated against
+    # the actual valid features supported by the target architecture.
+    #{ triple = "wasm32-unknown-unknown", features = ["atomics"] },
+]
+
+# This section is considered when running `cargo deny check advisories`
+# More documentation for the advisories section can be found here:
+# https://embarkstudios.github.io/cargo-deny/checks/advisories/cfg.html
+[advisories]
+# The path where the advisory database is cloned/fetched into
+db-path = "~/.cargo/advisory-db"
+# The url(s) of the advisory databases to use
+db-urls = ["https://github.com/rustsec/advisory-db"]
+# The lint level for security vulnerabilities
+vulnerability = "deny"
+# The lint level for unmaintained crates
+unmaintained = "warn"
+# The lint level for crates that have been yanked from their source registry
+yanked = "warn"
+# The lint level for crates with security notices. Note that as of
+# 2019-12-17 there are no security notice advisories in
+# https://github.com/rustsec/advisory-db
+notice = "warn"
+# A list of advisory IDs to ignore. Note that ignored advisories will still
+# output a note when they are encountered.
+ignore = [
+    #"RUSTSEC-0000-0000",
+]
```

Comment:
Is it possible to pass this via the command line so it can be an option into the action? I have seen, for example, an issue with a dependency that did not have an update path yet. You may want to still be able to run other checks without failing if there is a known vulnerability that you have no way of fixing yet.

Reply:
It looks like the answer is no. More specifically, it looks like `cargo-deny` only uses the `deny.toml` file for configuration. So our options appear to be:

1. Have a separate `deny.toml` in every repo (i.e. in the template)
2. Dynamically build a `deny.toml` based on environment variables
3. Add the dependency ignore to `deny.toml` in this repo
4. Do nothing and ignore the warnings.

I am personally leaning toward (3), with (2) as a possible necessary evil to keep a general, updated single source of truth.

Reply:
What if we do 1 but only in special circumstances? Like have the default here, but allow passing a different file path to the action if you need to override something _temporarily_ (ignore this vuln in this repo only).

We should be using dependabot alerts to inform us of vulnerable deps in all repos anyway, so we'll have an open issue to replace the dep which should also be our trigger to delete the override deny.toml.